packages/connect-examples/expo-playground/app/components/common/FirmwareFileUpload.tsx

```
@@ -0,0 +1,222 @@
+  const validateFile = useCallback(
+    (file: File): string | null => {
+      // 检查文件大小
+      if (file.size > maxSize) {
+        return `${t('components.firmwareFileUpload.fileSizeTooLarge')} (${formatFileSize(
+          maxSize
+        )})`;
+      }
+
+      // 检查文件格式
+      const fileExtension = '.' + file.name.split('.').pop()?.toLowerCase();
+      const acceptedExtensions = acceptedFormats.split(',').map(ext => ext.trim().toLowerCase());
```

Biome (1.9.4)

[error] Template literals are preferred over string concatenation.

Unsafe fix: Use a template literal.


(lint/style/useTemplate)